lib/logic.rb

```diff
@@ -0,0 +1,149 @@
+class Board
+  attr_accessor :board
+
+  def initialize
+    @turn = true
+    @board = [
+      [' ', ' ', ' '],
+      [' ', ' ', ' '],
+      [' ', ' ', ' ']
+    ]
+  end
+
+  def print_board
+    @board.each do |row|
+      puts row.to_s
+    end
+  end
+
+  def full?
+    @board.each do |row|
+      row.each do |value|
+        return false if value == ' '
+      end
+    end
+    true
+  end
+
+  def taken?(row, column)
+    if @board[row][column] != ' '
```

Comment:
It appears you are representing your board as a 2-dimensional array. Nothing wrong with it but can we make it simpler and better using a 1-dimensional array? See my comment in the @board variable above.